### DISCUSSION TITLE
MAC OS build error

### DISCUSSION THREAD

**Discussion by zhaorui233** [gem5/gem5#685] [Q&A]
![image](https://github.com/gem5/gem5/assets/52018861/b80001c9-db44-4e57-91c5-a2ada9a8ba5e)
I try to build gem5 on MACOS, and I got an error like this.
How to solve this error ?

**Comment by BobbyRBruce** [comment]
I'm going to need more information to be of help:

1. What build command are you using?
2. What version of gem5 are you using? Have you modified it in anyway?
3. What is your OS (i.e., the version)?
4. What ISA does your Mac use? X86 or Arm/AppleSilicon.
5. What compiler are you using (clang or gcc? What version?).

**Reply by zhaorui233** [reply]
1. scons build/X86/gem5.opt -j12
2. I'm using the latest version. I clone the gem5 project using 'git clone https://github.com/gem5/gem5' at 2024-01-08.  And I didn't change anything.
3. the OS version is: 14.2.1 (23C71)
4. CPU of my MAC is Apple M3 Pro
5. I don't know which compiler I'm using, I installed both clang and gcc on my MAC.
![image](https://github.com/gem5/gem5/assets/52018861/e1397b79-d78c-46b6-9beb-42036fb95fc5)

**Comment by hnpl** [comment]
Confusing linker error could be a result of not having enough memory during the linking phase. Please make sure that you're using >= 16GiB of RAM. Another option is to use gem5's `--limit-ld-memory-usage` compilation flag, e.g. `scons  build/ARM/gem5.opt -j4 --limit-ld-memory-usage`.

**Reply by zhaorui233** [reply]
Memory of my MAC is 36 GB. Is it enough? 
I got an error like this using the --limit-ld-memory-usage option.
![image](https://github.com/gem5/gem5/assets/52018861/9803ea6b-124b-4867-b13c-1ed77e5ed5cc)
